### PR TITLE
TST: Mock clipboard IO

### PR DESCRIPTION
**Pull request by TomAugspurger**
No idea if this will work on not.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
This failed.

Unless anyone objects, I'm going to just mock the actual clipboard layer part of things.

**Comment by jreback** [comment]
yeah the clipboard tests are annoying

File: pandas/tests/io/test_clipboard.py [hunk 1]
```diff
@@ -76,10 +76,41 @@ def df(request):
         raise ValueError
 
 
+# our local clipboard for tests
+_mock_data = {}
```
Comment (jreback) [inline pandas/tests/io/test_clipboard.py]: it’s really odd to keep state outside of this fixture when it’s a function level

why don’t u just make it a module level?
or it seems that function level might be ok with state inside

File: pandas/tests/io/test_clipboard.py [hunk 2]
```diff
@@ -76,10 +76,41 @@ def df(request):
...
+
+
+@pytest.fixture
+def mock_clipboard(mock, request):
+    def _mock_set(data):
+        _mock_data[request.node.name] = data
```
Comment (jreback) [inline pandas/tests/io/test_clipboard.py]: can u add a docstring

File: pandas/tests/io/test_clipboard.py [hunk 3]
```diff
@@ -76,10 +76,41 @@ def df(request):
...
+
 @pytest.mark.single
+@pytest.mark.clipboard
 @pytest.mark.skipif(not _DEPS_INSTALLED,
                     reason="clipboard primitives not installed")
+@pytest.mark.usefixtures("mock_clipboard")
 class TestClipboard(object):
+
+    @pytest.mark.clipboard
+    def test_mock_clipboard(self):
```
Comment (jreback) [inline pandas/tests/io/test_clipboard.py]: can’t u use monkey patch here?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by TomAugspurger** [comment]
Moved the mock clipboard to be function-scope, rather than global.

**Comment by jreback** [comment]
sure

**Pull request merged**